### PR TITLE
Add the ability to remove files when using file search

### PR DESCRIPTION
**Pull request by Stronkkey**
Implements [godot-proposals#7881](https://github.com/godotengine/godot-proposals/issues/7881)

### CODE REVIEW

File: editor/find_in_files.h [hunk 1]
```diff
@@ -177,13 +179,15 @@ class FindInFilesPanel : public Control {
 
 private:
 	void _on_result_found(String fpath, int line_number, int begin, int end, String text);
+	void _on_button_clicked(TreeItem *item, int column, int id, int mouse_button_index);
```
Comment (AThousandShips) [inline editor/find_in_files.h:182]:
```suggestion
	void _on_button_clicked(TreeItem *p_item, int p_column, int p_id, int p_mouse_button_index);
```
Arguments should be prefixed with `p_`

File: editor/find_in_files.cpp [hunk 2]
```diff
@@ -874,6 +897,20 @@ void FindInFilesPanel::_on_replace_all_clicked() {
 	emit_signal(SNAME(SIGNAL_FILES_MODIFIED), modified_files);
 }
 
+void FindInFilesPanel::_on_button_clicked(TreeItem *item, int column, int id, int mouse_button_index) {
```
Comment (AThousandShips) [inline editor/find_in_files.cpp:889]:
```suggestion
void FindInFilesPanel::_on_button_clicked(TreeItem *p_item, int p_column, int p_id, int p_mouse_button_index) {
```

File: editor/find_in_files.cpp [hunk 3]
```diff
@@ -549,6 +557,7 @@ void FindInFilesDialog::_bind_methods() {
 const char *FindInFilesPanel::SIGNAL_RESULT_SELECTED = "result_selected";
 const char *FindInFilesPanel::SIGNAL_FILES_MODIFIED = "files_modified";
 
+
```
Comment (AThousandShips) [inline editor/find_in_files.cpp:555]:
```suggestion
```

File: editor/find_in_files.h [hunk 4]
```diff
@@ -33,6 +33,8 @@
 
 #include "core/templates/hash_map.h"
 #include "scene/gui/dialogs.h"
+#include "scene/resources/placeholder_textures.h"
```
Comment (AThousandShips) [inline editor/find_in_files.h:36]:
```suggestion
```

File: editor/find_in_files.h [hunk 5]
```diff
@@ -33,6 +33,8 @@
...
+#include "scene/resources/texture.h"
```
Comment (AThousandShips) [inline editor/find_in_files.h:37]:
```suggestion
```

File: editor/find_in_files.cpp [hunk 6]
```diff
@@ -32,7 +32,9 @@
 
 #include "core/config/project_settings.h"
 #include "core/io/dir_access.h"
+#include "core/object/callable_method_pointer.h"
```
Comment (AThousandShips) [inline editor/find_in_files.cpp:35]:
```suggestion
```

File: editor/find_in_files.cpp [hunk 7]
```diff
@@ -32,7 +32,9 @@
...
 #include "core/os/os.h"
+#include "core/string/string_name.h"
```
Comment (AThousandShips) [inline editor/find_in_files.cpp:37]:
```suggestion
```

File: editor/find_in_files.cpp [hunk 8]
```diff
@@ -45,6 +47,7 @@
 #include "scene/gui/line_edit.h"
 #include "scene/gui/progress_bar.h"
 #include "scene/gui/tree.h"
+#include "scene/resources/texture.h"
```
Comment (AThousandShips) [inline editor/find_in_files.cpp:50]:
```suggestion
```

**Review by AThousandShips — COMMENTED** [review]
You have unrelated commits you need to remove from your contribution

File: editor/find_in_files.h [hunk 9]
```diff
@@ -177,13 +177,15 @@ class FindInFilesPanel : public Control {
...
 	void _on_result_found(String fpath, int line_number, int begin, int end, String text);
+	void _on_button_clicked(TreeItem *p_item, int p_column, int p_id, int p_mouse_button_index);
 	void _on_finished();
 	void _on_refresh_button_clicked();
 	void _on_cancel_button_clicked();
 	void _on_result_selected();
 	void _on_item_edited();
 	void _on_replace_text_changed(String text);
 	void _on_replace_all_clicked();
+	Texture2D* _get_remove_button_texture();
```
Comment (AThousandShips) [inline editor/find_in_files.h:188]:
```suggestion
```

File: servers/rendering/renderer_rd/effects/fsr2.cpp [hunk 10]
```diff
@@ -804,7 +804,9 @@ FSR2Effect::~FSR2Effect() {
 	RD::get_singleton()->free(device.linear_clamp_sampler);
 
 	for (uint32_t i = 0; i < FFX_FSR2_PASS_COUNT; i++) {
-		RD::get_singleton()->free(device.passes[i].pipeline.pipeline_rid);
+		if (device.passes[i].pipeline.pipeline_rid.is_valid()) {
+			RD::get_singleton()->free(device.passes[i].pipeline.pipeline_rid);
+		}
```
Comment (AThousandShips) [inline servers/rendering/renderer_rd/effects/fsr2.cpp:809]: You still have unrelated code, and another author not part of this, you will need to remove this code and reset the author of your commit with `git commit --reset-author`

**Comment by Stronkkey** [comment]
I messed up my commit history by syncing the repo. Will be reopened soon.

**Comment by Stronkkey** [comment]
I mean I'll make a new clean pull request

**Comment by Stronkkey** [comment]
[New pull request.](https://github.com/godotengine/godot/pull/82629)
